backend/fs-backend.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+import { mkdir, readFile, writeFile } from 'node:fs/promises'
+import path from 'node:path'
+
+import { checkKey } from './database.js'
+
+let dataFolder = './data'
+
+export async function initStorage (options: Object = {}): Promise<void> {
+  dataFolder = path.resolve(options.dirname)
+  await mkdir(dataFolder, { mode: 0o750, recursive: true })
+}
+
+// eslint-disable-next-line require-await
+export async function readString (key: string): Promise<string | void> {
+  checkKey(key)
+  return readFile(`${dataFolder}/${key}`)
```

Comment:
Please use the system-independent `path.join`.

I'm also not 100% sure this is safe, even with `checkKey`, because I'm not sure if specially encoded entities could encode `../` and somehow escape. Might be a question for @corrideat? This is why I used `path.resolve` in `throwIfFileOutsideDataDir`, as with `path.resolve` you can be 100% sure it's safe.

Reply:
1. `throwIfFileOutsideDataDir` seems like it would work. 
2. Re. system-independent `path.join`: `/` or `\\` should work as path separators in any system
3. specially encoded entities could encode `../`: I didn't test, but I'm guessing not, as Node.js should decode the URI. In any case, one could check the `key` doesn't contain any `%` (actually, it'd be better to have a whitelist of characters, like `/^[a-zA-Z0-9=+_-]$/`).

Reply:
I think we do want to allow support for foreign characters though, so I'm not sure about the whitelist. Sounds like the above code should work then...

Reply:
``Node `fs` methods don't seem to interpret strings as URIs:
![image](https://user-images.githubusercontent.com/64228468/230668239-5159d11e-5fbf-499b-9069-d54afaef96f6.png)

![image](https://user-images.githubusercontent.com/64228468/230667945-a89225a7-c9fa-461f-8faf-3aca3a52235b.png)

Reply:
OK, it's probably fine. If you're sure Windows is happy with `/`... we can stick with this